sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,38 @@
-    Pandas DataFrame method :meth:`pandas.DataFrame.count` implementation.
-
-    .. only:: developer
-
-        Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_count
-
-    Parameters
-    -----------
-    self: :class:`pandas.DataFrame`
-        input arg
-    axis:
-        *unsupported*
-    level:
-        *unsupported*
-    numeric_only:
-        *unsupported*
-
-    Returns
-    -------
-    :obj:`pandas.Series` or `pandas.DataFrame`
-            returns: For each column/row the number of non-NA/null entries. If level is specified returns a DataFrame.
-    """
-
-    _func_name = 'Method pandas.dataframe.count().'
-
-    if not isinstance(self, DataFrameType):
-        raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
-
-    if not (isinstance(axis, types.Omitted) or axis == 0):
-        raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
-
-    if not (isinstance(level, types.Omitted) or level is None):
-        raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
-
-    if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
-        raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
-
-    def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
-        result_data = []
-        result_index = []
-
-        for dataframe_item in self._data:
-            item_count = dataframe_item.count()
-            item_name = dataframe_item._name
-            result_data.append(item_count)
-            result_index.append(item_name)
-
-        return pandas.Series(data=result_data, index=result_index)
-
-    return sdc_pandas_dataframe_count_impl
+from sdc.datatypes.hpat_pandas_series_functions import TypeChecker
+
+def sdc_pandas_dataframe_reduce_columns(df, name, params):
+    saved_columns = df.columns
+    n_cols = len(saved_columns)
+    data_args = tuple('data{}'.format(i) for i in range(n_cols))
+    all_params = ['df']
+    for key, value in params:
+        all_params.append('{}={}'.format(key, value))
+    func_definition = 'def _reduce_impl({}):'.format(', '.join(all_params))
+    func_lines = [func_definition]
+    for i, d in enumerate(data_args):
+        line = '  {} = hpat.hiframes.api.init_series(hpat.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
+        func_lines.append(line.format(d + '_S', i))
+        func_lines.append('  {} = {}.{}()'.format(d + '_O', d + '_S', name))
+    func_lines.append('  data = np.array(({},))'.format(
+        ", ".join(d + '_O' for d in data_args)))
+    func_lines.append('  index = hpat.str_arr_ext.StringArray(({},))'.format(
+        ', '.join('"{}"'.format(c) for c in saved_columns)))
+    func_lines.append('  return hpat.hiframes.api.init_series(data, index)')
+    loc_vars = {}
+    func_text = '\n'.join(func_lines)
+
+    exec(func_text, {'hpat': sdc, 'np': numpy}, loc_vars)
```

Comment:
@Rubtsowa I think we should stick to the new name:
```suggestion
    exec(func_text, {'sdc': sdc, 'np': numpy}, loc_vars)
```
and use sdc.hiframes instead hpat.hiframes in the generated function text too.